src/net/tcp.ts

```diff
@@ -116,6 +111,18 @@ export class Server {
       // TODO: investigate which other values occur
       this.fulfillShutdown_(SocketCloseKind.UNKOWN);
     }
+
+    // Since there may be other calls to this provider in progress which will
+    // fail to resolve if we destroy the communication channel now, postpone
+    // doing so until the next iteration of the event loop.
+    setTimeout(() => {
```

Comment:
This seems incredibly fragile.  Is there a reason that this is actually safe?  Could we structure this to wait for a promise so that it's actually deterministic?